### PR TITLE
osd: set pg role more carefully

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by athanatos** [comment]
lgtm

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -3480,6 +3479,10 @@ void OSD::handle_pg_peering_evt(
       PG::_create(*rctx.transaction, pgid, pgid.get_split_bits(pp->get_pg_num()));
       PG::_init(*rctx.transaction, pgid, pp);
 
+      int role = osdmap->calc_pg_role(whoami, acting, acting.size());
+      if (!pp->is_replicated() && pgid.shard != whoami)
+	role = -1;
```
Comment (xiexingguo) [inline src/osd/OSD.cc]: I see you hard coded pg role to be -1, 0, 1 for ec pgs(https://github.com/ceph/ceph/pull/8834/commits/dfce95d124824b69a944983289be5630076d29fa#diff-fa6c2eba8356ae1442d1bf749beacfdfR3228), shall we do this here, too?

**Comment by liewegas** [comment]
repushed.. earlier version didn't even compile

**Comment by xiexingguo** [comment]
looks good now:-)

**Pull request merged**